common/diplomatic_actions/imperia_diplomatic_actions.txt

```diff
@@ -245,11 +233,214 @@ overlord_bankroll_protectorate = {
+		# The format here is as if scope:actor was asking if root wants a bankroll
+		accept_score = {
+			value = 0
+			add = {
+				desc = "DIPLOMATIC_ACCEPTANCE_BASE"
+				value = 5
+			}
+			add = {
+				desc = "DIPLOMATIC_ACCEPTANCE_RANK"
+				if = {
+					limit = {
+						scope:actor = {
+							country_rank < root.country_rank
+						}
+					}
+					value = -5
+				}
+				if = {
+					limit = {
+						scope:actor = {
+							country_rank > root.country_rank
+						}
+					}
+					value = 5
+				}
+			}
+			add = {
+				desc = "DIPLOMATIC_ACCEPTANCE_GDP"
+				value = "scope:actor.ai_gdp_comparison(root)"
+				max = 20
+				min = -20
+			}
+			if = {
+				limit = {
+					scaled_debt >= 0.05
+				}
+				add = {
+					desc = "DIPLOMATIC_ACCEPTANCE_DEBT"
+					value = scaled_debt
+					multiply = 30
+					min = 1
+				}
+			}
+			else_if = {
+				limit = {
+					scaled_gold_reserves >= 0.05
+				}
+				subtract = {
+					desc = "DIPLOMATIC_ACCEPTANCE_GOLD_RESERVES"
+					value = scaled_gold_reserves
+					multiply = 20
+					min = 1
+				}
+			}
+		}
+		# This is only used for determining the value of a bankroll in sways, so it follows different logic from normal action acceptance
+		# The format here is as if scope:actor was asking for a bankroll from root
+		junior_accept_score = {
+			value = 0
+			add = {
+				desc = "DIPLOMATIC_ACCEPTANCE_BASE"
+				value = -10
+			}
+			add = {
+				desc = "DIPLOMATIC_ACCEPTANCE_RANK"
+				if = {
+					limit = {
+						scope:actor = {
+							country_rank > root.country_rank
+						}
+					}
+					value = -5
+				}
+				if = {
+					limit = {
+						scope:actor = {
+							country_rank < root.country_rank
+						}
+					}
+					value = 5
+				}
+			}
+			subtract = {
+				value = 10
+				multiply = scope:actor.num_income_transfer_pacts
+			}
+			subtract = {
+				value = 10
+				multiply = num_income_transfer_pacts
+			}
+			add = {
+				desc = "DIPLOMATIC_ACCEPTANCE_GDP"
+				value = "scope:actor.ai_gdp_comparison(root)"
+				add = 2
+				multiply = -10
+				min = -100
+				max = 0
+			}
+			if = {
+				limit = {
+					scaled_debt >= 0.05
+				}
+				subtract = {
+					desc = "DIPLOMATIC_ACCEPTANCE_DEBT"
+					value = scaled_debt
+					multiply = 100
+					min = 1
+				}
+			}
+			else_if = {
+				limit = {
+					scaled_gold_reserves >= 0.05
+				}
+				add = {
+					desc = "DIPLOMATIC_ACCEPTANCE_GOLD_RESERVES"
+					value = scaled_gold_reserves
+					multiply = 25
+					min = 1
+				}
+			}
+		}
+		propose_score = {
+			value = 10
+		}
+		will_propose = {
+			scaled_debt <= 0
+			is_at_war = no
+			weekly_net_fixed_income > 0
+			OR = {
+				num_income_transfer_pacts < 1
+				AND = {
+					num_income_transfer_pacts < 2
+					income >= 200000
+				}
+				AND = {
+					num_income_transfer_pacts < 3
+					income >= 400000
+				}
+			}
+			OR = {
+				root = {
+					owes_obligation_to = scope:target_country
+				}
+				NOT = {
+					scope:target_country = {
+						owes_obligation_to = root
+					}
+				}
```

Comment:
Our version doesn't result in obligations so this part is not a valid reason to start bankrolling.